s3repo/model.py

```diff
@@ -378,9 +388,103 @@ def delete_package(self, package):
         """Delete a package from S3."""
         NotImplementedError("delete_package hasn't been implemented yet.")
 
+    def is_dir_or_file(self, path):
+        """Check if path is directory or file."""
+
+        list_parameters = {'Bucket': self.bucket.name,
+                           'Prefix': path}
+
+        try:
+            result = self.s3_client.list_objects_v2(**list_parameters)
+        except self.s3_client.exceptions.NoSuchBucket:
+            raise RuntimeError("No such bucket.")
+        contents = result.get('Contents')
+        # if contents is not None:
+        #     subfolders = 0
+        #     for fileobj in contents:
+        #         key = contents.get('Key')
+        #         tail = key[len(path):]
+        #         # if tail
+            
+        #     if tail != '' and tail[0] == '/':
+        #         return 'dir'
+        return 'file'
+
+    def get_directory(self, path):
+        """Get lists and metadata of files and folders within
+        directory from S3.
+        """
+
+        paginator = self.s3_client.get_paginator('list_objects')
+        list_parameters = {'Bucket': self.bucket.name,
```

Comment:
AFAIU if you add "Delimiter": '/', then you will only get the keys inside the directory (files in the "Key" section and directories in the "CommonPrefixes" section)
```
{'ResponseMetadata': {'HTTPStatusCode': 200, 'HTTPHeaders': {'server': 'nginx/1.16.1', 'date': 'Thu, 17 Jun 2021 09:05:03 GMT', 'content-type': 'application/xml', 'content-length': '800', 'connection': 'keep-alive', 'x-req-id': 'z8q33oq5', 'x-host': 'hb-front9'}, 'RetryAttempts': 0}, 'IsTruncated': False, 'Contents': [{'Key': 'Sources (3)', 'LastModified': datetime.datetime(2021, 6, 17, 9, 4, 44, 928000, tzinfo=tzutc()), 'ETag': '"975d93622e557d183efa334049dc4ee6"', 'Size': 1580, 'StorageClass': 'STANDARD'}, {'Key': 'test-dir', 'LastModified': datetime.datetime(2021, 6, 3, 9, 15, 41, 535000, tzinfo=tzutc()), 'ETag': '"d41d8cd98f00b204e9800998ecf8427e"', 'Size': 0, 'StorageClass': 'STANDARD'}], 'Name': 'tarantool-dev', 'Prefix': '', 'Delimiter': '/', 'MaxKeys': 1000, 'CommonPrefixes': [{'Prefix': 'check/'}, {'Prefix': 'test-dir/'}], 'EncodingType': 'url', 'KeyCount': 4}
```

Reply:
Done.